src/dxtbx/format/FormatNXmxDLS.py

```diff
@@ -13,36 +12,63 @@
 from dxtbx.format.FormatNXmx import FormatNXmx
 
 
-def get_bit_depth_from_meta(meta_file_name):
+def get_bit_depth_from_meta(meta_file_name: Path) -> int:
     with h5py.File(meta_file_name) as f:
-        return int(f["/_dectris/bit_depth_image"][()])
+        with contextlib.suppress(KeyError):
+            return int(f["/_dectris/bit_depth_image"][()])
 
+        # This might be a very old meta file, that only had 'datatype'?
+        if "datatype" in f:
+            unique_vals = np.unique(f["datatype"])
+            if not len(unique_vals) == 1:
+                raise RuntimeError(
```

Comment:
`sys.exit()`?

Reply:
https://github.com/cctbx/dxtbx/blob/88c7db30652aa15249e3837de8ff7db1f671caa5/src/dxtbx/format/FormatNXmxDLS.py#L79-L80 is what I was basing this on. This caught _all_ exceptions with a fallback, now it only catches a limited range (which doesn't catch e.g. AttributeError which is what was causing the problem before).

I should at least add a log warning that a fallback is being used.